openmdao/core/system.py

```diff
@@ -1792,7 +1797,13 @@ def update_src_indices(name, key):
                                            (self.msginfo, name, str(flat_src_indices),
                                             str(meta['flat_src_indices'])))
 
-                meta['src_indices'] = np.asarray(src_indices, dtype=INT_DTYPE)
+                if isinstance(src_indices, np.ndarray):
+                    contains_slice = [True if isinstance(i, slice) else False for i in src_indices]
```

Comment:
Since you do this check in multiple places, I'd recommend creating a function to do it.  Also, `contains_slice = any(isinstance(i, slice) for i in src_indices)` is cleaner.  It also gives you a simple bool instead of a list of bools that you have to iterate through again later.